Projects/UOContent/Items/Weapons/Abilities/Bladeweave.cs

```diff
@@ -0,0 +1,102 @@
+using Server.Mobiles;
+using System;
+using System.Collections.Generic;
+
+namespace Server.Items
+{
+    public class Bladeweave : WeaponAbility
+    {
+        private class BladeWeaveRedirect
+        {
+            public WeaponAbility NewAbility;
+            public int ClilocEntry;
+
+            public BladeWeaveRedirect(WeaponAbility ability, int cliloc)
+            {
+                NewAbility = ability;
+                ClilocEntry = cliloc;
+            }
+        }
+
+        private static readonly Dictionary<Mobile, BladeWeaveRedirect> _newAttack = new Dictionary<Mobile, BladeWeaveRedirect>();
+
+        public static bool BladeWeaving(Mobile attacker, out WeaponAbility a)
+        {
+            if (_newAttack.TryGetValue(attacker, out var bwr))
+            {
+                a = bwr.NewAbility;
+                return true;
+            }
+
+            a = null;
+            return false;
+        }
+
+        public override int BaseMana => 30;
+
+        public override bool OnBeforeSwing(Mobile attacker, Mobile defender)
+        {
+            if (!Validate(attacker) || !CheckMana(attacker, false))
```

Comment:
Let's fix the missing braces on all of the if statements. Sorry for not flagging them all before.